src/org/zaproxy/zap/extension/ascanrulesAlpha/AbstractAppFilePlugin.java

```diff
@@ -0,0 +1,142 @@
+                                        createTestablePath(baseUriPath));
+                } catch (URIException uEx) {
+                        if (LOG.isDebugEnabled()) {
+                                LOG.debug("An error occurred creating a URI for the: " + getName() + " scanner. " + uEx.getMessage(),
+                                                uEx);
+                        }
+                        return;
+                }
+                try {
+                        newRequest.getRequestHeader().setURI(elmahUri);
+                } catch (URIException uEx) {
+                        if (LOG.isDebugEnabled()) {
+                                LOG.debug("An error occurred setting the URI for a new request used by: " + getName() + " scanner. "
+                                                + uEx.getMessage(), uEx);
+                        }
+			return;
+                }
+                // Until https://github.com/zaproxy/zaproxy/issues/3563 is addressed track completed in Kb
+                // TODO change this when possible
+                synchronized (getKb()) {
+                    if (getKb().getBoolean(elmahUri, MESSAGE_PREFIX)) {
+                        return;
+                    }
+                    getKb().add(elmahUri, MESSAGE_PREFIX, Boolean.TRUE);
+                }
+                try {
+                        sendAndReceive(newRequest, false);
+                } catch (IOException e) {
+                        LOG.warn("An error occurred while checking [" + newRequest.getRequestHeader().getMethod() + "] ["
+                                        + newRequest.getRequestHeader().getURI() + "] for " + getName() + " Caught "
+                                        + e.getClass().getName() + " " + e.getMessage());
+                        return;
+                }
+                int statusCode = newRequest.getResponseHeader().getStatusCode();
+                if (statusCode == HttpStatusCode.OK) {
+                        raiseAlert(newRequest, getRisk(), "");
+                } else if (statusCode == HttpStatusCode.UNAUTHORIZED || statusCode == HttpStatusCode.FORBIDDEN) {
+                        raiseAlert(newRequest, Alert.RISK_INFO, getOtherInfo());
+                }
+	}
+
+	private String createTestablePath(String baseUriPath) {
+                String newPath = "";
+                if (baseUriPath.contains("/")) {
+                        if (baseUriPath.endsWith("/")) {
+                                newPath = baseUriPath + filename;
+                        } else {
+                                newPath = baseUriPath.substring(0, baseUriPath.lastIndexOf('/')) + "/" + filename;
+                        }
+                } else {
+                        newPath = baseUriPath + "/" + filename;
+                }
+                return newPath;
+        }
+	
+        private void raiseAlert(HttpMessage msg, int risk, String otherInfo) {
```

Comment:
Will these details always be applicable?

Reply:
Same as above.

Reply:
Ok